### DISCUSSION TITLE
Avengers Authentication using OAuthAPI

### DISCUSSION THREAD

**Discussion by filipecardoso0** [FEUP-MEIC-DS-2024-25/ai4sd#24] [Ideas]
Following professor Daniel suggestion, I am creating this discussion due to the ephemerality of Discord messages.
In order to contribute in a positive way to the overall avengers project, my group T03G03 decided that we could **integrate the external authentication API's of Google and Github** (known as OAuth).
Of course, this could only be done after the creation of the database. 
We have a suggestion however. Each avenger would have their own backend linked to their corresponding table(or collection I don't really know the terminology in terms of _Non Relational Database Models_) and they would retrieve the user details from a table/collection that will have all of this information, so each avenger could retrieve what they find useful out of it.
We are looking forward to hear what our fellow colleagues think about it. Let us know if you have any suggestion or if you would like to contribute to this asset as well, by implementing other external authentication API's as well (perhaps Uporto login would be nice).

**Comment by jmisidro** [comment]
I think it’s a great idea! As long as it’s implemented with Firestore, I’m all for it!

**Comment by ricardo-peralta-0000** [comment]
Hello. What's the state of the authentication for the avengers? Will it still be used or is it to remain as it is?
